### PR TITLE
THRIFT-5633: [build infra] add ubuntu jammy in build infra

### PR DESCRIPTION
**Pull request by Jimexist**
add ubuntu jammy in build infra

- [x] Did you create an [Apache Jira](https://issues.apache.org/jira/projects/THRIFT/issues/) ticket?  (not required for trivial changes)
- [x] If a ticket exists: Does your pull request title follow the pattern "THRIFT-NNNN: describe my issue"?
- [x] Did you squash your changes to a single commit?  (not required, but preferred)
- [x] Did you do your best to avoid breaking changes?  If one was needed, did you label the Jira ticket with "Breaking-Change"?
- [x] If your change does not involve any code, include `[skip ci]` anywhere in the commit message to free up build resources.

### CODE REVIEW

**Review by fishy — APPROVED** [review]
almost all travis steps failed 😞

the cross-test ones are swift, not sure if we recently merged a swift change broke it, or a new swift version broke it.

then ubsan failed because of ruby tests, one autotools failed because of cpp tests, the rest failed because of python tests.

none seems to be caused by ubuntu jammy

File: build/docker/ubuntu-jammy/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+ENV PATH /usr/local/lib/otp/bin:$PATH
+
+RUN apt-get install -y --no-install-recommends \
+  `# GlibC dependencies` \
+  libglib2.0-dev
+
+# golang
+ENV GOLANG_VERSION 1.19
+ENV GOLANG_DOWNLOAD_URL https://go.dev/dl/go$GOLANG_VERSION.linux-amd64.tar.gz
+ENV GOLANG_DOWNLOAD_SHA256 464b6b66591f6cf055bc5df90a9750bf5fbc9d038722bb84a9d56a2bea974be6
```
Comment (fishy) [inline build/docker/ubuntu-jammy/Dockerfile:150]: you would want to match this with recently merged https://github.com/apache/thrift/commit/32bd0bd950739ca048d10783a7617e317c8b9419, though.

Comment (Jimexist) [reply build/docker/ubuntu-jammy/Dockerfile:150]: Let me address in one shot in a subsequent pr

**Comment by Jimexist** [comment]
> 

yes i have been trying to resolve these during last few weeks.

   - > the cross-test ones are swift, not sure if we recently merged a swift change broke it, or a new swift version broke it.
   
      i guess due to uuid not properly working in swift, so is golang and rust, for which i tried to add in:
      - #2700 
      - #2695 
- for the ruby case i had no clue;
- for the java ssl errors, i had the impression that it was due to jdk bug but it was not a case because i had upgraded java to jdk 17
      - #2686 
      and the error persists; because we have java and kotlin build covered in GitHub action so it should be fine

my plan is to gradually cover more languages in GitHub action and then fix the cross test actions; for travis, that will be fixed later if i have more time

**Pull request merged**